src/util/message.h

```diff
@@ -44,8 +46,51 @@ struct messaget
     template <typename... Args>
     static void println(FILE *f, VerbosityLevel lvl, Args &&...args)
     {
-      if(lvl == VerbosityLevel::Error)
-        fmt::print(f, "ERROR: ");
+      /** 
+		* Newly added code to address issue #886.
+		* It prepends a well formatted timestamp and log level to each LOG message. 
+		* It is transparent to the existing logging function, i.e., log_message.
+		* TODO: It can be implemented using a separate function and invoked here and/or elsewhere when necessary. 		
+		*/
+      struct tm *
+        timestamp_struct; //A time structure consisting of 9 members, such as tm_sec, tm_min, tm_hour, tm_mday, tm_mon, tm_year, etc.
+      constexpr auto timestamp_format = "YYYY-MM-DD HH:MM:SS";
+      constexpr int timestamp_size =
+        std::char_traits<char>::length(timestamp_format) + 1;
```

Comment:
I suggest replacing these two lines with:
```
constexpr std::string timestamp_format {"YYYY-MM-DD HH:MM:SS"};
constexpr int timestamp_size = timestamp_format.size();
```

Reply:
@mikhailramalho Thank you for your suggestions. However, the `constexpr std::string timestamp_format {"YYYY-MM-DD HH:MM:SS"};` definition returns the following compile-time error. 
- error: the type ‘const string’ {aka ‘const std::__cxx11::basic_string<char>’} of ‘constexpr’ variable ‘timestamp_format’ is not literal

Although `constexpr` is efficient, it is usually problematic unless its compile-time memory allocation/deallocation is handled properly. Hence, I used `const` instead in the revised code.

Reply:
sure, that's fine, but my question still stands: why do we have two formats? Both `"YYYY-MM-DD HH:MM:SS"` and `"%Y-%m-%d %H:%M:%S"` are present in your code.

Reply:
@mikhailramalho I used the two formats for different purposes. The former is just a template of the actual appearance of the timestamp (e.g., 2023-03-08 21:15:00). This allows us to define the size of the timestamp and also to pass it as an argument in the `strftime` function. The latter is a character string specifying the format of conversion in the `strftime` function (refer https://en.cppreference.com/w/cpp/chrono/c/strftime). But, the latter does not allow us to compute the size of the timestamp.

Reply:
If I would suggest this way to accomodate both your remarks and mine: check the return value of `strftime` and if it 0 retry with a larger buffer. This has the following benefits over the current version:
1. we're not making assumptions on the formatting/size, which leads to
2. the time could be output in a locale-dependent way later on without changes
3. the time format string could optionally be overridden via an environment variable, allowing my use case above

If this way is chosen, I would recommend to keep the computed size of the formatted result or the buffer itself in a (static) member in order to avoid recomputing it on every log message.

Reply:
@fbrausse Thanks for your suggestion. I think your assumption is only partially correct. Because, the function of `strftime` is to convert the date/time information from a given calendar time to a character string according to a particular format string specified, e.g., `"%Y-%m-%d %H:%M:%S"`. This time format is locale-independent. It always produces the time in the form of `YYYY-MM-DD HH:MM:SS`. Furthermore, the format string is hardcoded, so it will not be overridden by an environment variable. So, we can assume that size of the resulting timestamp is fixed. However, we still define it using the string `"YYYY-MM-DD HH:MM:SS"`. This is because, the size of a character string is dependent on the architecture of the target machine.

I agree with your idea of defining the size as a static member variable.